fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/api/attachment/v1/AttachmentRegistry.java

```diff
@@ -114,6 +114,13 @@ public interface Builder<A> {
 		 */
 		Builder<A> persistent(Codec<A> codec);
 
+		/**
+		 * Declares that when a player dies and respawns, the attachments corresponding of this type should remain.
+		 *
+		 * @return the builder
+		 */
+		Builder<A> copyOnPlayerRespawn();
```

Comment:
This should only apply to serializable attachments.

Reply:
Why should it? Imagine data that records something about a player's current play session. It should clearly persist over respawns, but by definition it should be reset on disconnect.